### ISSUE TITLE
Version text within README leads to unexpected wheel name/version

### ISSUE THREAD

**Issue by cmditch** [pypa/setuptools-scm#776]
**tldr; Notice the mismatched wheel and tar version numbers.**
```
➜  my-package git:(main) git tag                     
v0.0.1
v0.0.2
v0.0.3
v0.0.4

➜  my-package git:(main) ls dist                     
total 32
drwxr-xr-x   4 cmditch  staff   128B Oct 31 22:13 .
drwxr-xr-x  16 cmditch  staff   512B Oct 31 22:13 ..
-rw-r--r--   1 cmditch  staff   6.5K Oct 31 22:13 my-package-0.0.4.tar.gz
-rw-r--r--   1 cmditch  staff   6.1K Oct 31 22:13 my-package-42.123.0-py3-none-any.whl

➜  my-package git:(main) cat README.md | grep Version
Version: 42.123.0
```

## Description

I have a repo with a `README.md` which includes the text `Version: 42.123.0` within it.

This version has nothing to do with the version of the package, it just happens to be in the README.

The latest tag on the repo is `v0.0.4`.

When building the package w/ `python -m build` a tar file is created with the name `my-package-0.0.4.tar.gz`, as expected.
But, a wheel file is produced matching the version within the README: `my-package-42.123.0-py3-none-any.whl`.

Deleting the `Version: 42.123.0` from the README fixes the issue. Also, this behavior only occurs if `Version: 42.123.0` is capitalized and separated with a colon. So `Version 42.123.0` and `version: 42.123.0` do not result in the above behavior.

**Is this the intended behavior or indeed a bug?**

**Comment by RonnyPfannschmidt** [comment]
Very interesting bug, that should not happen

**Comment by RonnyPfannschmidt** [comment]
Please run with `SETUPTOOLS_SCM_DEBUG=true` and share the repo if feasible

**Comment by RonnyPfannschmidt** [comment]
Closing as no debug details provided
